Registry/RegistryParse.py

```diff
@@ -1426,3 +1518,242 @@ def records(self):
         from warnings import warn
         warn("records is obsolete, use cells instead!")
         return self.cells()
+
+class HvLEBlock(RegistryBlock):
+    """
+    A HvLEBlock is the log entry in a new transaction log file.
+    It's length is multiple of 0x200.
+    """
+    def __init__(self, buf, offset, parent):
+        """
+        Constructor.
+        Arguments:
+        - `buf`: Byte string containing Windows Registry transaction log file.
+        - `offset`: The offset into the file-like object at which the block starts.
+        - `parent`: The parent block, which links to this block. The parent of the first HvLEBlock
+        should be the REGFBlock, and the parents of other HvLEBlocks should be the preceeding
+        HvLEBlocks.
+        """
+        super(HvLEBlock, self).__init__(buf, offset, parent)
+
+        _id = self.unpack_dword(0)
+        if _id != 0x454C7648:
+            raise ParseException("Invalid HvLE ID")
+
+        self._offset_next_hvle = self._offset + self.size()
+        self._marvin32seed = 0x82EF4D887A4E55C5
+
+    def __str__(self):
+        return "HvLE at 0x%x" % (self._offset)
+
+    def marvin32_hash(self, buf):
```

Comment:
do you have a source for this code, or did you write it yourself? if you got it elsewhere, thats fine, just want to be sure to include a reference. if its yours, nice work!